File: java-spring-boot/src/main/java/ar/sistemas360/examples/model/CrearComprobanteRequest.java
package ar.sistemas360.examples.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public record CrearComprobanteRequest(
    @JsonProperty("tipo_comprobante")
    String tipoComprobante,

    String concepto,

    LocalDate fecha,

    @JsonProperty("referencia_externa")
    String referenciaExterna,

    ClienteRequest cliente,

    List<ItemRequest> items,

    BigDecimal total,

    String moneda
) {
}

File: dotnet/console/Models/CrearComprobanteRequest.cs
using System.Text.Json.Serialization;

namespace Sistemas360Example.Models;

public sealed record CrearComprobanteRequest(
    [property: JsonPropertyName("tipo_comprobante")]
    string TipoComprobante,

    [property: JsonPropertyName("concepto")]
    string Concepto,

    [property: JsonPropertyName("fecha")]
    string Fecha,

    [property: JsonPropertyName("referencia_externa")]
    string ReferenciaExterna,

    [property: JsonPropertyName("cliente")]
    ClienteRequest Cliente,

    [property: JsonPropertyName("items")]
    IReadOnlyList<ItemRequest> Items,

    [property: JsonPropertyName("total")]
    decimal Total,

    [property: JsonPropertyName("moneda")]
    string Moneda
);
